File: docs/examples/python/challenges/get_challenge_info.py
"""
Get Challenge Information

Retrieves detailed information about a challenge including map, settings, and creator.

Usage:
    python get_challenge_info.py

Requirements:
    pip install requests
"""

import requests
import os
from datetime import datetime


def get_challenge_info(challenge_token, cookie):
    """
    Fetch and display challenge information.

    Args:
        challenge_token (str): The challenge token/ID
        cookie (str): Your _ncfa cookie value

    Returns:
        dict: Challenge information
    """
    try:
        url = f'https://www.geoguessr.com/api/v3/challenges/{challenge_token}'

        response = requests.get(url, cookies={'_ncfa': cookie})
        response.raise_for_status()

        challenge = response.json()

        # Display formatted challenge information
        print(f"\n🎮 Challenge Information\n")
        print(f"Map: {challenge['map']['name']}")
        print(f"Creator: {challenge['creator']['nick']} ({challenge['creator']['countryCode']})")
        print(f"Created: {datetime.fromisoformat(challenge['created'].replace('Z', '+00:00')).strftime('%Y-%m-%d %H:%M:%S')}")

        print(f"\n⚙️  Settings:")
        print(f"- Rounds: {challenge['roundCount']}")
        time_limit = 'None' if challenge['timeLimit'] == 0 else f"{challenge['timeLimit']}s"
        print(f"- Time Limit: {time_limit}")
        print(f"- Movement: {'❌ Disabled' if challenge['forbidMoving'] else '✅ Enabled'}")
        print(f"- Panning: {'❌ Disabled' if challenge['forbidRotating'] else '✅ Enabled'}")
        print(f"- Zooming: {'❌ Disabled' if challenge['forbidZooming'] else '✅ Enabled'}")

        print(f"\n📍 Map Details:")
        print(f"- Name: {challenge['map']['name']}")
        print(f"- Description: {challenge['map'].get('description', 'N/A')}")
        print(f"- Locations: {challenge['map'].get('coordinateCount', 'Unknown')}")

        if 'bounds' in challenge['map']:
            print(f"- Bounds: {challenge['map']['bounds']}")

        return challenge

    except requests.exceptions.HTTPError as e:
        print(f"❌ HTTP Error: {e}")
        if e.response.status_code == 403:
            print("Note: You may need to play the challenge first to view details")
        return None
    except requests.exceptions.RequestException as e:
        print(f"❌ Request Error: {e}")
        return None


def main():
    # Get credentials from environment variables
    cookie = os.getenv('GEOGUESSR_COOKIE')
    if not cookie:
        print("❌ Error: GEOGUESSR_COOKIE environment variable not set")
        print("Set it with: export GEOGUESSR_COOKIE='your_cookie_value'")
        return

    # Example challenge token
    challenge_token = input("Enter challenge token (or URL): ").strip()

    # Extract token from URL if full URL provided
    if '/' in challenge_token:
        challenge_token = challenge_token.split('/')[-1]

    # Fetch challenge info
    challenge = get_challenge_info(challenge_token, cookie)

    if challenge:
        print("\n✅ Successfully retrieved challenge information!")


if __name__ == '__main__':
    main()

File: examples/javascript/challenges/get-challenge-info.js
/**
 * Get Challenge Information
 *
 * Retrieves detailed information about a challenge including map, settings, and creator.
 *
 * Usage in browser console:
 * 1. Navigate to geoguessr.com (must be logged in)
 * 2. Paste this code into the console
 * 3. Call: getChallengeInfo('YOUR_CHALLENGE_TOKEN')
 */

async function getChallengeInfo(challengeToken) {
    try {
        const url = `https://www.geoguessr.com/api/v3/challenges/${challengeToken}`;

        const response = await fetch(url, {
            credentials: 'include'
        });

        if (!response.ok) {
            throw new Error(`HTTP ${response.status}: ${response.statusText}`);
        }

        const challenge = await response.json();

        // Display formatted challenge information
        console.log(`\n🎮 Challenge Information\n`);
        console.log(`Map: ${challenge.map.name}`);
        console.log(`Creator: ${challenge.creator.nick} (${challenge.creator.countryCode})`);
        console.log(`Created: ${new Date(challenge.created).toLocaleString()}`);
        console.log(`\n⚙️  Settings:`);
        console.log(`- Rounds: ${challenge.roundCount}`);
        console.log(`- Time Limit: ${challenge.timeLimit === 0 ? 'None' : challenge.timeLimit + 's'}`);
        console.log(`- Movement: ${challenge.forbidMoving ? '❌ Disabled' : '✅ Enabled'}`);
        console.log(`- Panning: ${challenge.forbidRotating ? '❌ Disabled' : '✅ Enabled'}`);
        console.log(`- Zooming: ${challenge.forbidZooming ? '❌ Disabled' : '✅ Enabled'}`);

        console.log(`\n📍 Map Details:`);
        console.log(`- Type: ${challenge.map.name}`);
        console.log(`- Description: ${challenge.map.description || 'N/A'}`);
        console.log(`- Locations: ${challenge.map.coordinateCount || 'Unknown'}`);

        // Return full challenge object
        return challenge;

    } catch (error) {
        console.error('❌ Error fetching challenge info:', error.message);
        return null;
    }
}

// Example usage:
// getChallengeInfo('6G9h2UPctmUmUtaa');

// Get info for current page challenge:
// const currentToken = window.location.pathname.split('/').pop();
// getChallengeInfo(currentToken);
